qiskit/backends/_qasmsimulator.py

```diff
@@ -139,6 +139,8 @@ def __init__(self, configuration=None):
         else:
             self._configuration = configuration
 
+    local_random = random.Random()
```

Comment:
Let's prefix this with: `self.local_random` :)

Reply:
The variable local_random is an instance variable for the QasmSimulator class.  When I prefix it with `self.` I get an error
`NameError: name 'self' is not defined`

Reply:
Ouch! You are right... could you move it above the `__init__(...)` method? It will increase readability a lot :)
```python
    local_random = random.Random()

    def __init__(self, configuration=None):
```
Thanks @hellmersjl3ds !

Reply:
Ok, my fault, I thought it was inside the `__init__()` function. The point here is that every instance of the QasmSimulator should has it's own  local copy of the Random object, so they don't interfere each other in case many instances of the simulator are running at the same time. Therefore, `local_random` should be an instance variable as you said, but it must be initialized into the `__init__(self)` method like this:
```python
   def __init__(self):
       ...
       self.local_random = random.Random()

```